### PR TITLE
enable more aggressive inlining of flows

### PR DESCRIPTION
**Pull request by lukevalenty**

### CODE REVIEW

File: include/flow/graph_builder.hpp [hunk 1]
```diff
@@ -149,7 +149,14 @@ struct graph_builder {
             constexpr auto built = build(v);
             static_assert(built.has_value(),
                           "Topological sort failed: cycle in flow");
-            return *built;
+
+            constexpr auto functionPtrs = built->functionPtrs;
+            constexpr auto size = functionPtrs.size();
+            constexpr auto name = built->name;
+
+            return [&]<std::size_t... Is>(std::index_sequence<Is...>) {
```
Comment (elbeno) [inline include/flow/graph_builder.hpp:157]: **thought**: I don't think the lambda needs to capture anything.

File: test/flow/graph_builder.cpp [hunk 2]
```diff
@@ -6,119 +6,140 @@
...
+[[maybe_unused]] constexpr auto milestone1 = flow::milestone<"milestone1">();
 
-constexpr auto a = flow::action<"a">([] { actual += "a"; });
-constexpr auto b = flow::action<"b">([] { actual += "b"; });
-constexpr auto c = flow::action<"c">([] { actual += "c"; });
-constexpr auto d = flow::action<"d">([] { actual += "d"; });
+[[maybe_unused]] constexpr auto a = flow::action<"a">([] { actual += "a"; });
+[[maybe_unused]] constexpr auto b = flow::action<"b">([] { actual += "b"; });
+[[maybe_unused]] constexpr auto c = flow::action<"c">([] { actual += "c"; });
+[[maybe_unused]] constexpr auto d = flow::action<"d">([] { actual += "d"; });
```
Comment (elbeno) [inline test/flow/graph_builder.cpp:17]: **question**: Are all these `[[maybe_unused]]`s needed, or are they an artifact of getting the tests to work that can now be removed?

File: include/flow/graph_builder.hpp (hunk 1 above)
Comment (lukevalenty) [reply include/flow/graph_builder.hpp:157]: tried without the capture, it needs it for the `constexpr` variables....even though they are `constexpr` :(

File: test/flow/graph_builder.cpp (hunk 2 above)
Comment (lukevalenty) [reply test/flow/graph_builder.cpp:17]: they are an artifact...will remove

**Review by elbeno — APPROVED** [review]

**Pull request merged**